### PR TITLE
FIX: Python - add metadata.quality.Parameter class; docstring and other formatting; change setup procedure

### PR DESCRIPTION
**Pull request by dmeaux**
**Fixes:**
- Added the `Parameter` class to the `metadata.quality` module.
- Added  type-hinting for return type of the `metadata.quality.Measure.parameter` attribute.
- Fixed the `metadata.service.OperationMetadata.parameters` attribute type-hinting to return a `Sequence`.
- Reformatted various docstrings to Google-style docstrings, which increases legibility in the docstrings and the online documentation.
- Updated the setup, development, test, documentation, and build processes to use a `pyproject.toml` file, modernizing the dependency management process.

### CODE REVIEW

File: geoapi/src/main/python/README.md [hunk 1]
```diff
@@ -1,43 +1,95 @@
-# GeoAPI Python REAME
+# GeoAPI Python README
 
 ## Static Type Checking
-Static type checking for GeoAPI's Pytnon implementation is accomplished using
-[mypy](https://mypy.readthedocs.io/en/stable/index.html).
+Static type checking for GeoAPI's Pytnon implementation is accomplished using [mypy](https://mypy.readthedocs.io/en/stable/index.html).
```
Comment (desruisseaux) [inline geoapi/src/main/python/README.md:4]: We can keep this on two lines in the `.md` file. They will still appear as a single paragraph in the rendered page. When using Markdown, all consecutive lines are in the same paragraph. A real line feed is done only when there is a blank line in the `.md` file.

The reason for splitting on many lines is to make the `.md` file easier to read, but also for smaller diff in commits. For better change isolation in commit, I often split the lines at natural boundaries such as a coma or a period.

File: geoapi/src/main/python/opengis/__init__.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 """
 
 __package_name__ = "opengis"
-__version__ = 4.0
-__author__ = "Martin Desruisseaux (Geomatys), David Meaux (Geomatys)"
+__version__ = "4.0a1"
+__author__ = "Martin Desruisseaux(Geomatys), David Meaux (Geomatys)"
```
Comment (desruisseaux) [inline geoapi/src/main/python/opengis/__init__.py:27]: Missing space before (Geomatys).

File: geoapi/src/main/python/opengis/py.typed
Comment (desruisseaux) [inline geoapi/src/main/python/opengis/py.typed:1]: Do we need this empty file?

Comment (dmeaux) [reply geoapi/src/main/python/opengis/py.typed:1]: Yes, it's for static type checkers. See [PEP 561 - Packaging Type Information](https://peps.python.org/pep-0561/#packaging-type-information)

File: geoapi/src/main/python/opengis/__init__.py (hunk 2 above)
Comment (dmeaux) [reply geoapi/src/main/python/opengis/__init__.py:27]: Sorry, I thought I had fixed that.

**Pull request merged**